src/sv-pipeline/04_variant_resolution/scripts/StitchFragmentedCNVs.java

```diff
@@ -0,0 +1,242 @@
+
+/** Read a VCF, and try to stitch together adjacent copy-number variations.
+ * Eligible Records (which we call "stitchable") must meet certain criteria as specified by the
+ * isStitchable method of the StitchableIterator.
+ * If two stitchables overlap in a weird way that seems wrong to me (see line 54), and all their
+ * samples have identical genotypes, we can replace the first one by adding on the interval covered
+ * by the second one.
+ */
+public class StitchFragmentedCNVs {
+    private static final VCFParser.ByteSequence END = new VCFParser.ByteSequence("END");
+    private static final VCFParser.ByteSequence SVLEN = new VCFParser.ByteSequence("SVLEN");
+    private static final double PAD_FACTOR = .2;
+    private static final int MAX_PAD = 200000;
+    private static final String GZ = ".gz";
+
+    public static void main( final String[] args ) {
+        try ( final OutputStream os = createOutputStream(args[1]) ) {
+            try ( final VCFParser vcfParser = new VCFParser(args[0]) ) {
+                while ( vcfParser.hasMetadata() ) {
+                    vcfParser.nextMetaData().write(os);
+                }
+                final StitchableIterator sItr = new StitchableIterator(vcfParser);
+                VCFParser.Record stitchableRecord;
+                while ( (stitchableRecord = sItr.nextSubject(os)) != null ) {
+                    findExtension(stitchableRecord, sItr);
+                    stitchableRecord.write(os);
+                }
+            }
+        } catch ( final IOException ioe ) {
+            throw new VCFParser.MalformedVCFException("can't write output to " + args[1], ioe);
+        }
+    }
+
+    private static void findExtension( final VCFParser.Record stitchable,
+                                       final StitchableIterator sItr ) throws IOException {
+        final int startPos = stitchable.getPosition();
+        final VCFParser.ByteSequence endField = stitchable.getInfoField(END);
+        final int originalEndPos = endField.asInt(); // can't be null -- checked in isStitchable
+        int endPos = originalEndPos;
+        int pad = Math.min(MAX_PAD, (int)(PAD_FACTOR * (endPos - startPos)));
+        final int padStartPos = Math.max(1, startPos - pad);
+        int padEndPos = endPos + pad;
+
+        // sItr.hasNext returns false at EOF, or when the next record is too far away to overlap the subject
+        while ( sItr.hasNext() ) {
+            final VCFParser.Record record = sItr.next();
+            final int startPos2 = record.getPosition();
+            final int endPos2 = record.getInfoField(END).asInt(); // can't be null -- checked in isStitchable
+            final int pad2 = Math.min(MAX_PAD, (int)(PAD_FACTOR * (endPos2 - startPos2)));
+            final int padStartPos2 = Math.max(1, startPos2 - pad2);
+            final int padEndPos2 = endPos2 + pad2;
+            final int overlap = endPos - startPos2; // this seems clearly wrong. need clarification.
```

Comment:
Pretty sure that this is what's written.  (See line 118 in the bash script.)
But now that I think about it, I think maybe it's ok.  I was worried that overlap could be negative for disjoint intervals, but I see now that we're assured that they don't "underlap" by too much since the padded intervals overlapped.  It seems to be a fix that avoids merging events that overlap by too much, where too much is defined by the minimum padding.